### ISSUE TITLE
There should be no syntax highlighting on braces that don't contain a comma 

### ISSUE THREAD

**Issue by KotlinIsland** [fish-shell/fish-shell#7963]
> If there is no "," or variable expansion between the curly braces, they will not be expanded:

\- [the docs](https://fishshell.com/docs/current/#expand-brace)

```fish
>echo foo{bar}
foo{bar}
```
![image](https://user-images.githubusercontent.com/65446343/116496834-45649c80-a8e9-11eb-8334-83b14d1607d3.png)

**Comment by KotlinIsland** [comment]
Same thing with "~foo", the tilda is highlighted but is a literal character.

**Comment by faho** [comment]
>Same thing with "~foo", the tilda is highlighted but is a literal character.

Not really. Fish still tries to do tilde-expansion here, and when you have a user named "foo" it'll expand to that user's home directory. I'm pretty sure we don't want to do that lookup in the background before you're running the command, so that bit doesn't seem fixable.

Plus, if you did copy that to another system and it did have a "foo" user it would behave differently.

Highlighting the braces differently is sensible but not super important.

**Comment by KotlinIsland** [comment]
Okay, that makes sense. I wouldn't mind looking into it, could you point me in the right direction?

**Comment by ridiculousfish** [comment]
[Here's the highlighting logic](https://github.com/fish-shell/fish-shell/blob/bcbfd70d413155152ce1261f925a62b678c83d32/src/highlight.cpp#L670), and [here's the expansion logic](https://github.com/fish-shell/fish-shell/blob/9d696ba7d27d62af20c6085b9378ad600a92d1d0/src/common.cpp#L1426). We don't have any sort of ast for expanding individual arguments so it's somewhat ad-hoc.
